modules/swagger-codegen/src/main/resources/Java/ApiClient.mustache

```diff
@@ -421,10 +421,10 @@ public class ApiClient {
       throw new ApiException(500, "unknown method type " + method);
     }
 
-    if(response.getClientResponseStatus() == ClientResponse.Status.NO_CONTENT) {
+    if(response.getStatusInfo() == ClientResponse.Status.NO_CONTENT) {
```

Comment:
This is incorrect. The ```response.getStatusInfo()``` returns an object implementing ```Response.StatusType``` interface, and not a ```ClientResponse.Status``` Enum, so the comparison will always fail. Because ```ClientResponse.Status``` also implements ```Response.StatusType``` interface, the correct fix is  ```response.getStatusInfo().getStatusCode() == ClientResponse.Status.NO_CONTENT.getStatusCode()```, which will compare the numeric values from the objects.